doc/langdef.md

```diff
@@ -1203,1342 +1193,925 @@ names are like `Europe/Paris`, `CET`, or `US/Central`.
+
+**Negation (-)** \- Takes a numeric value (int or double) as input and returns
+its negated value.
+
+**Signatures:**
+
+*   `-int -> int`
+*   `-double -> double`
+
+**Examples:**
+
+```
+-5    // -5
+-3.14 // -3.14
+```
+
+**Addition (+)** \- Adds two numeric values or concatenates two strings, bytes,
+or lists.
+
+**Signatures:**
+
+*   Numeric addition
+    *   `int + int -> int`
+    *   `uint + uint -> uint`
+    *   `double + double -> double`
+*   Time and duration addition
+    *   `google.protobuf.Timestamp + google.protobuf.Duration ->
+        google.protobuf.Timestamp`
+    *   `google.protobuf.Duration + google.protobuf.Timestamp ->
+        google.protobuf.Timestamp`
+    *   `google.protobuf.Duration + google.protobuf.Duration ->
+        google.protobuf.Duration`
+*   Concatenation
+    *   `string + string -> string`
+    *   `bytes + bytes -> bytes`
+    *   `list(A) + list(A) -> list(A)`
+
+**Examples:**
+
+```
+1 + 2 // 3
+3.14 + 1.59 // 4.73
+"Hello, " + "world!" // "Hello, world!"
+[1] + [2, 3] // [1, 2, 3]
+duration('1m') + duration('1s') // duration('1m1s')
+timestamp('2023-01-01T00:00:00Z')
+  +   duration('24h') // timestamp('2023-01-02T00:00:00Z')
+```
+
+**Subtraction (-)** \- Subtracts two numeric values or calculates the duration
+between two timestamps.
+
+**Signatures:**
+
+*   Numeric subtraction
+    *   `int - int -> int`
+    *   `uint - uint -> uint`
+    *   `double - double -> double`
+*   Time and duration subtraction
+    *   `google.protobuf.Timestamp - google.protobuf.Timestamp ->
+        google.protobuf.Duration`
+    *   `google.protobuf.Timestamp - google.protobuf.Duration ->
+        google.protobuf.Timestamp`
+    *   `google.protobuf.Duration - google.protobuf.Duration ->
+        google.protobuf.Duration`
+
+**Examples:**
+
+```
+5 - 3 // 2
+10.5 - 2.0 // 8.5
+duration('1m') - duration('1s') // duration('59s')
+timestamp('2023-01-10T12:00:00Z')
+  -   timestamp('2023-01-10T00:00:00Z') // duration('12h')
+```
+
+**Division (/)** \- Divides two numeric values.
+
+**Signatures:**
+
+*   `int / int -> int`
+*   `uint / uint -> uint`
+*   `double / double -> double`
+
+**Examples:**
+
+```
+10 / 2 // 5
+7.0 / 2.0 // 3.5
+```
+
+#### Comparison Operators
+
+**Equality (==)** \- Compares two values of the same type and returns `true` if
+they are equal, and `false` otherwise
+
+**Signatures:**
+
+*   `A == A -> bool` (where A can be any comparable type)
+
+**Examples:**
+
+```
+1 == 1 // true
+"hello" == "world" // false
+bytes('hello') == b'hello' // true
+duration('1h') == duration('60m') // true
+```
+
+**Inequality (\!=)** \- Takes two values of the same type and returns `true` if
+they are not equal, and `false` otherwise.
+
+**Signatures:**
+
+*   `A != A -> bool` (where A can be any comparable type)
+
+**Examples:**
+
+```
+1 != 2     // true
+"a" != "a" // false
+3.0 != 3.1 // true
+```
+
+**Less Than or Equal To (\<=)** \- Compares two values and returns `true` if the
```

Comment:
Should this include a note about cross-numeric type comparisons?

Reply:
I'll add a note at the top of the Comparisons section about cross-numeric type comparisons at runtime. Good call.